packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart

```diff
@@ -270,43 +270,47 @@ abstract class AbstractVerbHandler implements VerbHandler {
   ///
   /// Additionally, this function removes the OTP from the keystore to prevent
   /// its reuse.
-  Future<bool> isOTPValid(String? otp) async {
-    if (otp == null) {
+  Future<bool> isPasscodeValid(String? passcode) async {
+    if (passcode == null) {
       return false;
     }
-    // Check if user has configured SPP(Semi-Permanent Pass-code).
+    // 1. Check if user has configured an SPP(Semi-Permanent Pass-code).
     // If SPP key is available, check if the otp sent is a valid pass code.
     // If yes, return true, else check it is a valid OTP.
-    String sppKey =
-        'private:spp.${OtpVerbHandler.otpNamespace}${AtSecondaryServerImpl.getInstance().currentAtSign}';
-    if (!keyStore.isKeyExists(sppKey)) {
+    String passcodeKey = OtpVerbHandler.passcodeKey(passcode, isSpp: true);
+    if (!keyStore.isKeyExists(passcodeKey)) {
       // if new SPPKey does not exist in keystore, check for SPP data against legacy SPP key
       // New SPP key has __otp namespace, legacy key does NOT have any namespace
-      sppKey =
+      passcodeKey =
           'private:spp${AtSecondaryServerImpl.getInstance().currentAtSign}';
     }
     try {
-      AtData? sppAtData = await keyStore.get(sppKey);
+      AtData? sppAtData = await keyStore.get(passcodeKey);
       // SPP has a special key so we have to check the value that was stored
       // (which is the actual SPP)
       // By comparison, OTPs are stored with the key being ${OTP}.__otp@alice
       // i.e. the OTP is part of the key, and the stored data is irrelevant
-      if (sppAtData?.data?.toLowerCase() == otp.toLowerCase()) {
-        return true;
+      if (sppAtData?.data?.toLowerCase() == passcode.toLowerCase()) {
+        if (SecondaryUtil.isActiveKey(sppAtData)) {
```

Comment:
This is one part of the actual change for this PR - add a check to ensure the spp hasn't expired